### PR TITLE
[dotnet] Make it possible to specify the registrar using a 'Registrar' property in MSBuild.

### PR DESCRIPTION
**Pull request by rolfbjarne**
However, passing --registrar:... to mtouch/mmp will override any MSBuild property.

### CODE REVIEW

File: tests/ComputeRegistrarConstant.targets [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+			<IsDynamicRegistrar Condition="'$(ComputedPlatform)' == 'iPhoneSimulator' And '$(Registrar)' == ''">true</IsDynamicRegistrar>
+			<IsDynamicRegistrar Condition="'$(Registrar)' == 'dynamic'">true</IsDynamicRegistrar>
+		</PropertyGroup>
+
+		<PropertyGroup Condition="'$(_PlatformName)' == 'macOS' Or $(TargetFramework.EndsWith('-macos')) Or '$(_PlatformName)' == 'MacCatalyst' Or $(TargetFramework.EndsWith('-maccatalyst'))">
+			<IsDynamicRegistrar Condition="'$(Configuration)' == 'Debug'">true</IsDynamicRegistrar>
+		</PropertyGroup>
+
+		<PropertyGroup Condition="'$(HasDynamicRegistrar)' == 'true'">
+			<IsDynamicRegistrar>$(DefineConstants);DYNAMIC_REGISTRAR</IsDynamicRegistrar>
```
Comment (Therzok) [inline tests/ComputeRegistrarConstant.targets:14]: Should this be `<DefineConstants>`?

File: tests/ComputeRegistrarConstant.targets [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+	<Target Name="ComputeRegistrarConstant" BeforeTargets="BeforeBuild">
+		<PropertyGroup Condition="'$(_PlatformName)' == 'iOS' Or $(TargetFramework.EndsWith('-ios')) Or '$(_PlatformName)' == 'tvOS' Or $(TargetFramework.EndsWith('-tvos'))">
+			<IsDynamicRegistrar Condition="'$(ComputedPlatform)' == 'iPhoneSimulator' And '$(Registrar)' == ''">true</IsDynamicRegistrar>
+			<IsDynamicRegistrar Condition="'$(Registrar)' == 'dynamic'">true</IsDynamicRegistrar>
+		</PropertyGroup>
+
+		<PropertyGroup Condition="'$(_PlatformName)' == 'macOS' Or $(TargetFramework.EndsWith('-macos')) Or '$(_PlatformName)' == 'MacCatalyst' Or $(TargetFramework.EndsWith('-maccatalyst'))">
+			<IsDynamicRegistrar Condition="'$(Configuration)' == 'Debug'">true</IsDynamicRegistrar>
```
Comment (Therzok) [inline tests/ComputeRegistrarConstant.targets:10]: Should this be set only if not already set?

Comment (rolfbjarne) [reply tests/ComputeRegistrarConstant.targets:10]: This is just a local variable really, and only for tests, so it doesn't matter really.

File: tests/ComputeRegistrarConstant.targets (hunk 1 above)
Comment (rolfbjarne) [reply tests/ComputeRegistrarConstant.targets]: Yes, fixed

File: tests/ComputeRegistrarConstant.targets (hunk 2 above)
Comment (Therzok) [reply tests/ComputeRegistrarConstant.targets:10]: Just noticed it was a test targets :D

**Review by chamons — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Review by mauroa — APPROVED** [review]

**Pull request merged**